app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/AddressLookupRoutingController.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers
+
+import javax.inject.Inject
+import play.api.i18n.Lang
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.SelfEmploymentDataKeys.addressIdKey
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.config.AppConfig
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.httpparser.addresslookup.GetAddressLookupDetailsHttpParser.InvalidJson
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.httpparser.addresslookup.PostAddressLookupHttpParser.PostAddressLookupSuccessResponse
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.httpparser.addresslookup._
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.{AddressLookupConnector, IncomeTaxSubscriptionConnector}
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.models.BusinessAddressModel
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.services.AuthService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+
+import scala.concurrent.ExecutionContext
+
+class AddressLookupRoutingController @Inject()(mcc: MessagesControllerComponents,
+                                               authService: AuthService,
+                                               addressLookupConnector: AddressLookupConnector,
+                                               incomeTaxSubscriptionConnector: IncomeTaxSubscriptionConnector
+                                              )(implicit val ec: ExecutionContext, val appConfig: AppConfig) extends FrontendController(mcc) {
+
+  def initialiseAddressLookupJourney(): Action[AnyContent] = Action.async { implicit request =>
+    authService.authorised() {
+      implicit val lang: Lang = mcc.messagesApi.preferred(request).lang
+      val continueUrl = appConfig.incomeTaxSubscriptionSelfEmployedFrontendBaseUrl + routes.AddressLookupRoutingController.addressLookupRedirect(None).url
+      addressLookupConnector.initialiseAddressLookup(continueUrl) map (
```

Comment:
The initialise connector method is broken (not your controller code) because the json is being created through Json.toJson() instead of Json.parse(). I don't know where else to comment about this being broken.

Reply:
Done